### PR TITLE
Build worker.js from typescript src files

### PR DESCRIPTION
**Pull request by antiphoton**
Previously, the `worker.js` has been a single file containing many functions.

This PR creates a typescript project, which builds `worker.js`, making is possible to do type checking and unit testing.

### CODE REVIEW

File: cloudflare_worker/README.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ limitations under the License.
    1. The terminal will interactively ask for the value of the secret.
       Put the private key in JWK format here.
 
-1. `wrangler publish`
+1. Run `sh publish.sh` to build and deploy the worker online.
```
Comment (twifkak) [inline cloudflare_worker/README.md:51]: You use `sh` here but `#!/bin/bash` at the top of `publish.sh`. To address this inconsistency I would add the executable bit to publish.sh and then change the message here to "Run `./publish.sh` to build...".

Alternatively, just change the `sh` here or the `bash` in `publish.sh`.

File: cloudflare_worker/README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ limitations under the License.
    1. The terminal will interactively ask for the value of the secret.
```
Comment (twifkak) [inline cloudflare_worker/README.md:48]: Add a step above line 24 to `git clone` this repo and cd into cloudflare_worker.

File: cloudflare_worker/worker/karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  config.set({
+    basePath: '',
+    frameworks: ['jasmine'],
+    files: [
+      'dist/**/*.test.js'
+    ],
+    colors: true,
+    logLevel: config.LOG_INFO,
+    autoWatch: false,
+    browsers: ['ChromeHeadless'],
```
Comment (twifkak) [inline cloudflare_worker/worker/karma.conf.js:27]: Why is this line necessary? Do we have code that uses the DOM? I would think some sort of node test runner would be sufficient (and faster).

File: cloudflare_worker/worker/rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import typescript from '@rollup/plugin-typescript';
+
+const testSrcFiles = glob.sync('src/**/*.test.ts');
+
+export default [
+  {
+    input: 'src/index.ts',
+    output: {
+      dir: 'dist',
+      format: 'es'
```
Comment (twifkak) [inline cloudflare_worker/worker/rollup.config.js:27]: Maybe add a short comment saying what 'es' is. (Ditto 'cjs' below.)

File: cloudflare_worker/worker/src/bindings.d.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export {};
+
+declare global {
+  const PRIVATE_KEY_JWK: string
+  const OCSP: KVNamespace
+}
```
Comment (twifkak) [inline cloudflare_worker/worker/src/bindings.d.ts:22]: (nit/opt) Add a newline for consistency with other files.

**Review by twifkak — APPROVED** [review]
Nice to see the move to TypeScript! Just a few small suggestions.

File: cloudflare_worker/publish.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (banaag) [inline cloudflare_worker/publish.sh:1]: Consider dockerizing this.  Could be a TODO for now.

Comment (antiphoton) [reply cloudflare_worker/publish.sh:1]: Created cl/195126851

File: cloudflare_worker/README.md (hunk 1 above)
Comment (antiphoton) [reply cloudflare_worker/README.md]: Done by adding executable bit.

File: cloudflare_worker/worker/src/bindings.d.ts (hunk 5 above)
Comment (antiphoton) [reply cloudflare_worker/worker/src/bindings.d.ts]: Done

File: cloudflare_worker/README.md (hunk 2 above)
Comment (antiphoton) [reply cloudflare_worker/README.md:56]: Done

File: cloudflare_worker/worker/karma.conf.js (hunk 3 above)
Comment (antiphoton) [reply cloudflare_worker/worker/karma.conf.js:27]: The [worker runtime](https://developers.cloudflare.com/workers/runtime-apis/web-standards#web-global-apis) is similar to WorkerGlobalScope in browser, which provides some global functions not available in nodejs.

File: cloudflare_worker/worker/rollup.config.js (hunk 4 above)
Comment (antiphoton) [reply cloudflare_worker/worker/rollup.config.js]: Comment added. The format is now changed to the more appropriate format 'iife'.
